sympy/combinatorics/perm_groups.py

```diff
@@ -5217,12 +5216,8 @@ class SymmetricPermutationGroup(Basic):
 
     """
 
-    def __new__(cls, deg):
-        deg = _sympify(deg)
-        obj = Basic.__new__(cls, deg)
-        obj._deg = deg
-        obj._order = None
-        return obj
+    def __init__(self, *args, **kwargs):
+        self._order = factorial(self.args[0])
```

Comment:
Why this instead of the lazy computation upon a `__contains__` request?

Reply:
I can revert it for sure. Just seemed to make more sense to compute it once and for all rather than checking every time. But no strong feelings there...

Reply:
(I know it is computed once in the other setup as well.)